docs: move data.fields line in README

frapan opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

frapan commented:
Fixes #252

climba03003 reviewed (changes requested):
`(available only AFTER the stream is consumed)`
The above sentence is not always true. I do not want to give wrong explanation to the client just to reduce the issue.

Instead, I would like someone to explain more on how `stream` or `busboy` operate and create a Q&A section.

e.g.
`busboy` consume the `multipart` in serial order (stream), the **order** of form field is `VERY IMPORTANT` to how `fastify-multipart` can display the fields to you. We would recommend you to place the value fields first before any of the file fields. It can ensure your `fields` will be accessible before start consuming any file.

darkgl0w reviewed (changes requested):
> Note: if the file stream that is provided by data.file is not consumed, like in the example below with the usage of pump, the promise will not be fulfilled at the end of the multipart processing. This behavior is inherited from @fastify/busboy.

There is a note about this in the `README.md` just after the example

I would suggest that you add another note with the explaining sentence proposed by @climba03003 instead.

frapan commented:
Thank you for your explanations.

I reverted to the original file and added the note of @climba03003 with a small addition:

> **Note** about `data.fields`: `busboy` consumes the multipart in serial order (stream). Therefore, the order of form fields is *VERY IMPORTANT* to how `fastify-multipart` can display the fields to you.
> We would recommend you to place the value fields first before any of the file fields.
> It can ensure your fields will be accessible before start consuming any file.
> If you cannot control the order of the placed fields, be sure to read `data.fields` *AFTER* consuming the stream, or it will contain only the fields parsed at that moment.

I also realized that the following note was duplicated:

> Note: if the file stream that is provided by data.file is not consumed, like in the example below with the usage of pump, the promise will not be fulfilled at the end of the multipart processing. This behavior is inherited from @fastify/busboy.

So I deleted the one that was in the wrong (imho) position.

What do you think?

mccare commented:
On a related note, the documentation here https://github.com/fastify/busboy#busboy-special-events mentions that you can also just use `file.resume()`. If that is correct, I would also add it to the documentation here. WDYT?

climba03003 commented:
> On a related note, the documentation here https://github.com/fastify/busboy#busboy-special-events mentions that you can also just use `file.resume()`. If that is correct, I would also add it to the documentation here. WDYT?

I think it is a separate issue.

climba03003 commented:
Can you kindly help to check the sentence for us?
cc @Fdawgs

climba03003 reviewed (approved):
LGTM.

Fdawgs commented on README.md:
```diff
@@ -70,10 +70,12 @@ fastify.listen(3000, err => {
 
-You can also pass optional arguments to `@fastify/busboy` when registering with Fastify. This is useful for setting limits on the content that can be uploaded. A full list of available options can be found in the [`@fastify/busboy` documentation](https://github.com/fastify/busboy#busboy-methods).
+**Note** about `data.fields`: `busboy` consumes the multipart in serial order (stream). Therefore, the order of form fields is *VERY IMPORTANT* to how `fastify-multipart` can display the fields to you.
+We would recommend you to place the value fields first before any of the file fields.
```
```suggestion
We would recommend you place the value fields first before any of the file fields.
```

Fdawgs commented on README.md:
```diff
@@ -70,10 +70,12 @@ fastify.listen(3000, err => {
-You can also pass optional arguments to `@fastify/busboy` when registering with Fastify. This is useful for setting limits on the content that can be uploaded. A full list of available options can be found in the [`@fastify/busboy` documentation](https://github.com/fastify/busboy#busboy-methods).
+**Note** about `data.fields`: `busboy` consumes the multipart in serial order (stream). Therefore, the order of form fields is *VERY IMPORTANT* to how `fastify-multipart` can display the fields to you.
+We would recommend you to place the value fields first before any of the file fields.
+It can ensure your fields will be accessible before start consuming any file.
```
```suggestion
It will ensure your fields are accessible before it starts consuming any files.
```

Fdawgs commented on README.md:
```diff
@@ -70,10 +70,12 @@ fastify.listen(3000, err => {
+**Note** about `data.fields`: `busboy` consumes the multipart in serial order (stream). Therefore, the order of form fields is *VERY IMPORTANT* to how `fastify-multipart` can display the fields to you.
+We would recommend you to place the value fields first before any of the file fields.
+It can ensure your fields will be accessible before start consuming any file.
+If you cannot control the order of the placed fields, be sure to read `data.fields` *AFTER* consuming the stream, or it will contain only the fields parsed at that moment.
```
```suggestion
If you cannot control the order of the placed fields, be sure to read `data.fields` *AFTER* consuming the stream, or it will only contain the fields parsed at that moment.
```

mcollina commented:
@Fdawgs are you ok now?

Fdawgs commented:
> @Fdawgs are you ok now?

All gravy now
